src/VirtoCommerce.Platform.Web/Controllers/Api/AuthorizationController.cs

```diff
@@ -84,6 +87,15 @@ public async Task<ActionResult> Exchange(OpenIdConnectRequest openIdConnectReque
                     });
                 }
 
+
+                // check and expire user password if needed
+                var lastPasswordChangeDate = user.LastPasswordChangedDate ?? user.CreatedDate;
```

Comment:
Please move the code for evaluation the PasswordExpired to this place, and do not call update for user.
https://github.com/VirtoCommerce/vc-platform/blob/master/src/VirtoCommerce.Platform.Web/Security/CustomUserManager.cs#L287